src/character.cpp

```diff
@@ -3090,6 +3090,27 @@ void Character::on_move( const tripoint_abs_ms &old_pos )
     }
 }
 
+units::volume Character::get_total_volume() const
+{
+    item_location wep = get_wielded_item();
+    units::volume wep_volume = wep ? wep->volume() : 0_ml;
+    // Note: Does not measure volume of worn items that do not themselves contain anything
+    return get_base_volume() + volume_carried() + wep_volume;
```

Comment:
volume_carried() here doesn't seem appropriate, unless I'm reading it wrong wearing a giant empty rucksack counts as 0, while a snack bar in my jeans pocket apparently makes the space I'm in more cramped?

Reply:
That is the intention. Using worn clothes' volume would be wildly inappropriate, since (depending on the clothing in question) it would add negligible amounts of volume to your person. A wetsuit is 5L, but it's definitely not going to add 5L of *volume* to your person when being worn.

There is room for more nuance here, like separating out rigid and non-rigid items and materials.

Reply:
volume_carried() is wildly inappropriate too imo

Reply:
How? If you're carrying a 500L fridge and you try to step into a vehicle it's... just not gonna work? Same thing if you're wearing a huge box backpack.

Reply:
> Same thing if you're wearing a huge box backpack.

volume_carried() doesn't count that as volume unless it's full which was my original point...

Reply:
It does? In the tests we use a giant novelty backpack with 80L of volume capacity. Putting it on and putting a 10L rock into it causes an open tile to become cramped for movement purposes. It increases the character's volume from 70L to above 75L (where the aisle starts being cramped). If it counted the full 80L then our total volume would be 70+80 = 150L, and we couldn't move through the aisle at all. But we can, and it's just cramped.

Reply:
That's my point... the novelty backpack should be counting the full 80L (or whatever volume takes up when empty) bc it's not surrounding the player like a wetsuit. At the very least you should be counting BELTED stuff.
It's also pretty silly that like a flak jacket with plates would count the plates volume but an empty power armour suit wouldn't count as anything.